routes/web.php

```diff
@@ -17,6 +17,8 @@
 */
 
 Route::get('/', [PublicController::class, 'index'])->name('home');
+Route::get('/admin/posts/{post}/view', [PostController::class, 'view'])->name('posts.view');
+Route::resource('/admin/posts', PostController::class);
```

Comment:
This line already provides posts view route as well so line 20 is not needed use `php artisan route:list` to see what the url is etc. Also there seems to be no way to get tho that route since you didnt update posts.index view